### PR TITLE
add lamb optimizer

### PR DESCRIPTION
**Pull request by L1aoXingyu**
为 eager 和 graph 增加 lamb optimizer interface

原始公式如下

image

第一步 normalize gradients 可以在 clip grad 中实现。

### CODE REVIEW

**Comment by L1aoXingyu** [comment]
image

image

File: oneflow/api/python/functional/dispatch_stateful_ops.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+#include <memory>
```
Comment (MARD1NO) [inline oneflow/api/python/functional/dispatch_stateful_ops.cpp:17]: remove

File: python/oneflow/nn/optimizer/lamb.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        parameters: Union[Iterator[Parameter], List[Dict]],
+        lr: float = 0.001,
+        betas: Tuple[float, float] = (0.9, 0.999),
+        eps: float = 1e-08,
+        weight_decay: float = 0,
+        adam_w_mode: bool = True,
+        do_bias_correction: bool = True,
+        amsgrad: bool = False,
+    ):
+        if amsgrad:
```
Comment (MARD1NO) [inline python/oneflow/nn/optimizer/lamb.py:117]: 加个todo吧

**Review by MARD1NO — APPROVED** [review]

File: oneflow/user/kernels/model_update_kernel_util.cu [hunk 3]
```diff
@@ -447,53 +451,62 @@ template struct AdagradUpdateKernelUtil<DeviceType::kCUDA, double, double>;
...
+                     float bias_correction2_val, const float* learning_rate_ptr,
+                     const float* bias_correction1_ptr, const float* bias_correction2_ptr,
+                     const T* scale_by_ptr, const int64_t* skip_if, const float16* model_diff,
+                     T* adam_diff, T* model, T* m, T* v, T* norm_buffer
+) {
   LambUpdateKernelUtil<DeviceType::kCUDA, T, half>::Update(
-      stream, n, scale, l1, l2, beta1, beta2, epsilon, weight_decay, learning_rate, scale_by_ptr,
-      skip_if, reinterpret_cast<const half*>(model_diff), adam_diff, model, m, v, norm_buffer,
-      beta1_t, beta2_t);
+      stream, n, scale, l1, l2, beta1, beta2, epsilon, weight_decay, learning_rate_val, do_bias_correction, bias_correction1_val, bias_correction2_val, learning_rate_ptr, bias_correction1_ptr, bias_correction2_ptr, scale_by_ptr,
```
Comment (wyg1997) [inline oneflow/user/kernels/model_update_kernel_util.cu:508]: 这个 of_format 不会做分行吗？

**Review by wyg1997 — APPROVED** [review]
